back/app/policies/idea_internal_comment_policy.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class IdeaInternalCommentPolicy < ApplicationPolicy
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user  = user
+      @scope = scope
+    end
+
+    def resolve
+      scope.where(post_id: Pundit.policy_scope(user, Idea))
+    end
+  end
+
+  def create?
+    active_admin_or_moderator?
+  end
+
+  def children?
+    active_admin_or_moderator?
+  end
+
+  def show?
+    active_admin_or_moderator?
+  end
```

Comment:
Not sure. I imagined it was the FE that filtered the marked as 'deleted' comments. I will investigate.

Reply:
Answer: Yes. This is (currently) just a copy of the `comments` mark as deleted functionality, where the comment will still be returned in an index response, for example, but will have `"publication_status": "deleted"` and `"body_multiloc": null,` (`body_text` in the case of `internal_comments`).

So, if a user inspects the response, they can see a comment was deleted, but not what the comment body was.

Reply:
✅ Resolved: see [comment](https://github.com/CitizenLabDotCo/citizenlab/pull/5079#discussion_r1227742001)

Reply:
Since we realized that 'soft' deletion is needed to ensure the integrity of threaded comments, I have simply added a simple `internal_comment_serializer_spec.rb` that tests the body text and author data are both `nil` for an `internal_comment` that is marked as deleted.

I will also add this to the list for the ticket of tech debt for `comments` (adding such a serializer spec).